Feature/13865914 add drag and drop on photo list to order and send order to app

ericcleao opened the pull request.

ericcleao committed:
feature/13865914 Add drag and drop handler and touch helper

- Add drag and drop handler interface
- Add drag and drop touch helper
- Implement methods "get movement flags", "on move" and "on child draw"
on "drag and drop touch helper" that extends item touch helper callback

Signed-off-by: Eric C. Leão <ericcleao@gmail.com>

ericcleao committed:
feature/13865914 Add drag and drop implementation to camera fragment and camera adapter

- Create an alpha effect when user drags the image tha hideds the delete button and set alpha to 0.75
- Allign the two camera item designs to be equals

Signed-off-by: Eric C. Leão <ericcleao@gmail.com>

ericcleao committed:
feature/13865914 Implement "drag and drop touch helper" on card shon taken picture

- Add item touch helper on the card show taken pictures adapter
- Add vibrate to on item selected methods

Signed-off-by: Eric C. Leão <ericcleao@gmail.com>

ericcleao committed:
feature/13865914 Fix minors design problems

- Add item touch helper at card show taken pictures adapter
-

Signed-off-by: Eric C. Leão <ericcleao@gmail.com>

ericcleao committed:
feature/13865914 Add new parameter to onSaved method from "on saved listener" listener

- New paremeter "current images" that sends the current list status

Signed-off-by: Eric C. Leão <ericcleao@gmail.com>

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/CardShowTakenPictureViewImagesAdapter.java:
```diff
@@ -2,54 +2,77 @@
+    private void configureDefaultConstraintLayoutTouchListener(ItemViewHolder itemViewHolder) {
+        itemViewHolder.mServiceInspectionsFormFilledRecycleItemBinding.cardShowTakenPictureViewGeneralCircularImageView.setOnLongClickListener(
+                view -> {
+                    if (!mView.canEditState) {
```
create a boolean method with negation name to avoid use of "!",

ericcleao replied:
Fixed

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/CardShowTakenPictureViewImagesAdapter.java:
```diff
@@ -129,7 +152,19 @@ public int getItemCount() {
+    @Override
+    public void onViewMoved(int oldPosition, int newPosition) {
+        CardShowTakenImage targetCardShowTakenImage = mCurrentCardShowTakenImageList.get(oldPosition);
+        mCurrentCardShowTakenImageList.remove(oldPosition);
```
add space between variables initialization and methods calls

ericcleao replied:
Added

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/CardShowTakenPictureViewImagesAdapter.java:
```diff
@@ -2,54 +2,77 @@
-        this.mOriginalTempCardShowTakenImageList = new ArrayList<>();
-        this.mCardShowTakenImageListAsAdded      = new ArrayList<>();
-        this.mCardShowTakenImageListAsRemoved    = new ArrayList<>();
+        this.mCurrentCardShowTakenImageList      = new ArrayList<>(0);
```
remove "this" if has "m" pattern on global variables

ericcleao replied:
Fixed

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/camera/CameraPhotosAdapter.java:
```diff
@@ -136,7 +153,21 @@ private boolean isNotLoading() {
+    public void onViewMoved(int oldPosition, int newPosition) {
+        if (newPosition < mPhotos.size() && oldPosition < mPhotos.size()) {
+            CameraPhoto targetCameraPhoto = mPhotos.get(oldPosition);
+            mPhotos.remove(oldPosition);
```
add space between variables initialization and methods calls

ericcleao replied:
Fixed

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/res/layout/camera_photo_preview_dialog.xml:
```diff
@@ -10,31 +11,37 @@
-            android:layout_alignParentTop="true"
-            android:layout_alignParentEnd="true"
-            android:background="@drawable/shape_round_black"
+            android:id="@+id/close_preview_image"
```
add more specific id related with view name

ericcleao replied:
Fixed

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/res/layout/card_show_taken_picture_preview_dialog.xml:
```diff
@@ -22,6 +22,7 @@
             android:id="@+id/preview_image"
```
add more specific id related with view name

denisvieiradev commented on cardshowviewtakenpicturesview/src/test/java/br/com/stant/libraries/cardshowviewtakenpicturesview/CardShowTakenPictureViewImagesAdapterTest.java:
```diff
@@ -41,7 +41,7 @@
 //        @Test
 //        public void It_should_add_correct_picture(){
-//            String result = cardShowTakenPictureViewImagesAdapter.getData().get(0);
+//            String result = cardShowTakenPictureViewImagesAdapter.getCurrentImages().get(0);
```
commented code ? is better remove

denisvieiradev commented on sample/src/main/java/br/com/stant/libraries/card_show_taken_pictures_view/MainActivity.java:
```diff
@@ -31,28 +31,35 @@ protected void onCreate(Bundle savedInstanceState) {
-        mBinding.cardShowViewTakenPicturesView.setBackgroundColor(ContextCompat.getColor(getApplicationContext(), R.color.colorPrimary));
-        mBinding.cardShowViewTakenPicturesView.enableSaveOnlyMode(ContextCompat.getDrawable(getBaseContext(), R.drawable.ic_cam), "oii",
-                ContextCompat.getDrawable(getBaseContext(), R.drawable.ic_close_dialog), "Tchau");
+        mBinding.cardShowViewTakenPicturesView.setStrokeColor(ContextCompat.getColor(getApplicationContext(),
```
On this class, i believe that we create methods to test contexts, for not get very free . Create a method with your context and call here .

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/CardShowTakenPictureViewImagesAdapter.java:
```diff
@@ -141,15 +176,32 @@ public int getItemCount() {
+            setImageBitmapToImageView(this.mServiceInspectionsFormFilledRecycleItemBinding.cardShowTakenPictureViewGeneralCircularImageView,
+                    cardShowTakenImage, 8);
 
             this.mServiceInspectionsFormFilledRecycleItemBinding.setCardShowTakenImage(cardShowTakenImage);
```
remove "this"

ericcleao replied:
Fixed

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/CardShowTakenPictureViewImagesAdapter.java:
```diff
@@ -141,15 +176,32 @@ public int getItemCount() {
+        public void onItemClear() {
+            mServiceInspectionsFormFilledRecycleItemBinding.cardShowTakenPictureContainerConstraintLayout.setAlpha(1);
+            mServiceInspectionsFormFilledRecycleItemBinding.cardShowTakenPicturesItemDeleteIconContainerCardView
+                    .setVisibility(mView.getActualCardState() == CardShowTakenPictureStateEnum.EDIT ? View.VISIBLE: View.GONE);
```
create a method to ternary condition to become more explanatory

ericcleao replied:
Created

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/CardShowTakenPictureViewImagesAdapter.java:
```diff
@@ -141,15 +176,32 @@ public int getItemCount() {
+
+        @Override
+        public void onItemSelected() {
+            VibratorUtils.vibrate(mView.getContext(), 400);
```
create a variable to describe "400" constant and the programmer not need see method declaration

ericcleao replied:
Created

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/camera/CameraPhotosAdapter.java:
```diff
@@ -164,12 +195,23 @@ public void fileNotFound() {
 
+        @Override
+        public void onItemSelected() {
+            VibratorUtils.vibrate(mContext, 400);
```
create a variable to describe number constant and the programmer not need see method declaration

ericcleao replied:
Created

ericcleao committed:
feature/13865914 Fix PR comments

-

Signed-off-by: Eric C. Leão <ericcleao@gmail.com>

ericcleao committed:
feature/13865914 Update gradle version

- Version code updated from 24 to 25
- Version name updated from 2.8.0 to 2.9.0

Signed-off-by: Eric C. Leão <ericcleao@gmail.com>

ericcleao committed:
feature/13865914 Add enable drag and drop mode option to make this feature optional

Signed-off-by: Eric C. Leão <ericcleao@gmail.com>

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/CardShowTakenPictureViewImagesAdapter.java:
```diff
@@ -129,27 +144,62 @@ public int getItemCount() {
 
-            setImageBitmapToImageView(this.mServiceInspectionsFormFilledRecycleItemBinding.cardShowTakenPictureViewGeneralCircularImageView, cardShowTakenImage, 8);
+            setImageBitmapToImageView(mServiceInspectionsFormFilledRecycleItemBinding.cardShowTakenPictureViewGeneralCircularImageView,
+                    cardShowTakenImage, 8);
```
what is 8 number mean ? create a descriptive variable

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/camera/CameraPhotosAdapter.java:
```diff
@@ -17,23 +18,29 @@
     private boolean mLoading;
+    private ItemTouchHelper mTouchHelper;
+
+    private static final int vibrationDuration = 400;
```
is a constant ? change to capitalize with underline and put near of other

ericcleao committed:
feature/13865914 Fix PR coments

- Make variables more redables

Signed-off-by: Eric C. Leão <ericcleao@gmail.com>
